sql/hive/src/test/scala/org/apache/spark/sql/hive/HiveExternalCatalogVersionsSuite.scala

```diff
@@ -267,7 +273,7 @@ object PROCESS_TABLES extends QueryTest with SQLTestUtils {
       case NonFatal(_) => Nil
     }
     versions
-      .filter(v => v.startsWith("3") || !TestUtils.isPythonVersionAtLeast38())
+      .filter(v => v.startsWith("3") && isPythonVersionAtLeast37)
```

Comment:
Should we strictly restrict the test python version to be greater than 3.7? I manually tested 3.6+ Spark 3.3, also succeed